invenio_preservation_sync/models.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Preservation-Sync is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Models for Preservation Sync integration."""
+
+import uuid
+from enum import Enum
+
+from invenio_db import db
+from invenio_i18n import lazy_gettext as _
+from invenio_webhooks.models import Event
+from sqlalchemy.dialects import mysql, postgresql
+from sqlalchemy_utils.models import Timestamp
+from sqlalchemy_utils.types import ChoiceType, JSONType, UUIDType
+
+PRESERVATION_STATUS_TITLES = {
+    "PRESERVED": _("Preserved"),
+    "PROCESSING": _("Processing"),
+    "FAILED": _("Failed"),
+    "DELETED": _("Deleted"),
+}
+
+PRESERVATION_STATUS_ICON = {
+    "PRESERVED": "check icon",
+    "PROCESSING": "spinner loading icon",
+    "FAILED": "times icon",
+    "DELETED": "times icon",
+}
+
+PRESERVATION_STATUS_COLOR = {
+    "PRESERVED": "positive",
+    "PROCESSING": "warning",
+    "FAILED": "negative",
+    "DELETED": "negative",
+}
+
+
+class PreservationStatus(str, Enum):
+    """Constants for possible status of a preservation."""
+
+    __order__ = "PRESERVED PROCESSING FAILED DELETED"
+
+    PRESERVED = "P"
+    """Release was successfully processed and published."""
+
+    PROCESSING = "I"
+    """Release is still being processed."""
+
+    FAILED = "F"
+    """Release processing has failed."""
+
+    DELETED = "D"
+    """Release has been deleted."""
+
+    def __init__(self, value):
+        """Hack."""
```

Comment:
Why is this needed? Perhaps add an explanation in the comments

Reply:
Copied from the _invenio-github_ module, but I didn't find an explanation and removing it didn't cause any issues.